include/dr/shp/distributed_vector.hpp

```diff
@@ -158,7 +158,14 @@ struct distributed_vector {
 
   distributed_vector(std::initializer_list<T> init)
       : distributed_vector(init.size()) {
-    shp::copy(rng::begin(init), rng::end(init), begin());
+    // shp::copy(rng::begin(init), rng::end(init), begin());
```

Comment:
Bug here is that copy does not work because of issue described in [this comment](https://github.com/oneapi-src/distributed-ranges/pull/60#issuecomment-1447527300)

Another issue I am facing is that sometimes one test (part_not_aligned) adds 9 instead of 10 into one field when running on a PVC.
```
[ RUN      ] TransformTest/1.part_not_aligned
/nfs/site/home/lslusarc/src/distributed-ranges/test/gtest/shp/transform.cpp:64: Failure
Value of: equal(b, typename TestFixture::LocalVec{9, 9, 9, 9, 9, 11, 12, 13, 9})
  Actual: false (
    [9, 9, 9, 9, 9, 10, 12, 13, 9]
    [9, 9, 9, 9, 9, 11, 12, 13, 9]
  )
Expected: true
[  FAILED  ] TransformTest/1.part_not_aligned, where TypeParam = sycl::_V1::usm_allocator<unsigned long long, (sycl::_V1::usm::alloc)2, 0ul> (18 ms)
```
I shall report these issues as bugs tomorrow. + exclusive scan problem

Reply:
@BenBrock: Unaligned requires cross-GPU references, and that gets into coherency issues. What is the SHP memory model?

Reply:
With `shp` I'm assuming that peer-to-peer access is enabled and that GPUs can access each others' memory in a cache-coherent manner.  I believe this should be the case if you use USM shared on a multi-PVC system.  @jngkim maybe you can comment on the cross-GPU PVC memory model?